flax/core/spmd.py

```diff
@@ -32,6 +32,8 @@ def get_pspec(sharding_names, sharding_rules = None) -> PartitionSpec:
     return PartitionSpec(*from_sharding_rules(sharding_names, rules))
   return PartitionSpec(*sharding_names)
 
+def _apply_sharding(value, sharding):
+  return jax.jit(lambda x: x, out_shardings=sharding)(value)
```

Comment:
Should we wrap this line in?:
```python
with jax.disable_jit(False):
  ...
```
so that `_apply_sharding` works even if inside a `jax.disable_jit(True)`?